sklearn/decomposition/sparse_pca.py

```diff
@@ -185,6 +191,63 @@ def transform(self, X, ridge_alpha='deprecated'):
         U /= s
         return U
 
+    def get_explained_variance(self, components, X):
+        '''
+            Get the explained variance from the principal components of the
+            data.
+
+            Parameters
+            ----------
+            components : 2D np.ndarray
+                the principal components of shape p x m
+                where m is the number of samples and p the number of
+                components
+            X : 2D np.ndarray
+                The feature vector. Must be shape m x n where m is the number
+                of samples and n is the dimension of the incoming feature
+                vector
+
+            Notes
+            -----
+            The variance ratio may not be computed. The main reason is that we
+            do not know what the total variance is since we did not compute all
+            the components.
+        '''
+        # requires the mean to be subtracted
+        X = X - np.mean(X, axis=0)
+        # the number of samples
+        n_samples = X.shape[0]
+        n_components = components.shape[0]
+        unit_vecs = components.copy()
+        unit_vecs = unit_vecs/np.linalg.norm(components, axis=1)[:, np.newaxis]
+
+        # same shape as X but only fill first i components
+        # Algorithm, as we compute the variance for each projection,
+        # we subtract the variance from components in the direction of previous
+        # axes
+        # the projection corrected vectors
+        proj_corrected_vecs = np.zeros_like(components)
```

Comment:
too much verbose. I think you can remove first and last lines

Reply:
removed verbosity to all lines mentioned